src/Sylius/Bundle/PromotionBundle/Doctrine/ORM/CouponRepository.php

```diff
@@ -22,12 +22,40 @@ class CouponRepository extends EntityRepository implements CouponRepositoryInter
     /**
      * {@inheritdoc}
      */
+    public function countCouponsByCodeLength($codeLength)
+    {
+        $query = $this->createQueryBuilder('o');
+
+        $count = (int) $query->select($query->expr()->count('o'))
+            ->where($query->expr()->eq($query->expr()->length('o.code'), $codeLength))
+            ->getQuery()
+            ->getSingleScalarResult()
+        ;
+
+        return $count;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
     public function createQueryBuilderWithPromotion($promotionId)
     {
-        return
-            $this->createQueryBuilder('o')
-                ->where('o.promotion = :promotionId')
-                ->setParameter('promotionId', $promotionId)
-            ;
+        return $this->createQueryBuilder('o')
+            ->where('o.promotion = :promotionId')
+            ->setParameter('promotionId', $promotionId)
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function findCouponsByCodeLength($codeLength)
+    {
+        $query = $this->createQueryBuilder('o');
```

Comment:
Please rename these variables to `$queryBuilder`. `query` means something else and can be confusing.